tests/tests_within_docker

```diff
@@ -0,0 +1,282 @@
+#!/bin/bash
+
+# This script is run within a docker container that has access
+# to GKE credentials and configured clients.
+
+# 1. Deploys the sidecar injector
+# 2. Deploys a test app (echo server), which is configured to have
+#    secretless broker (SB) injected. The injected SB is configured via a CRD
+#    (custom resource definition). SB is configured to proxy http requests to
+#    the echo server, adding in a basic auth header.
+# 3. A request is made to the echo server with SB as the proxy, the test passes
+#    if a basic auth header (inserted by SB) is detected in the response from
+#    the echo server.
+
+set -exu
+set -o pipefail
+
+. ./utils
+
+# setup environment variables
+log=log.txt
+: >${log}
+export SECRETLESS_CRD_SUFFIX=${SECRETLESS_CRD_SUFFIX-}
+HELM_DEPLOYMENT="sci-${UNIQUE_TEST_ID}"
+
+# Clean up when script completes
+function cleanup() {
+    announce 'Wrapping up and removing test environment'
+
+    # Delete image from GCR
+    gcloud container images delete --force-delete-tags -q  "${SIDECAR_IMAGE}"
+
+    # Delete sci helm deployment
+    helm delete --purge "${HELM_DEPLOYMENT}"
+
+    # Delete k8s resources namespace
+    kubectl delete --wait=true --ignore-not-found=true namespace "${TEST_NAMESPACE}"
+    kubectl delete --wait=true --ignore-not-found=true namespace "i-${TEST_NAMESPACE}"
+    kubectl delete --wait=true --ignore-not-found=true crd "configurations.secretless${SECRETLESS_CRD_SUFFIX}.io" &
+
+    kubectl delete --wait=true --ignore-not-found=true service secretless-broker secretless-broker-external &
+
+    kubectl delete --wait=true --ignore-not-found=true clusterrole secretless-crd &
+    kubectl delete --wait=true --ignore-not-found=true rolebinding secretless-crd &
+    kubectl delete --wait=true --ignore-not-found=true clusterrolebinding secretless-crd &
+    kubectl delete --wait=true --ignore-not-found=true serviceaccount secretless-crd &
+
+    kubectl delete --wait=true --ignore-not-found=true deployment sb-sci-echoserver &
+    kubectl delete --wait=true --ignore-not-found=true pods -l app=sb-sci-echoserver &
```

Comment:
shouldn't we have a wait at the end of all these background execs ?

Reply:
Yes! Done.